lib/msal-react-native-android-poc/SampleApp/App.js

```diff
@@ -28,38 +28,95 @@ class App extends Component {
     super(props);
     this.state = {
       isSignedIn: false,
+      isSharedDevice: false,
       scopesValue: "",
-      aUsername: null,
-      aTenantId: null,
-      aIdToken: null,
-      aAuthority: null,
-      aId: null,
-      resultMap: null
+      URLValue: "https://graph.microsoft.com/v1.0/me",
+      account: null,
+      graphResult: null
     };
-    this.getAccount();
+  }
+
+  async componentDidMount() {
+    try {
+      await this.getAccount();
+      await this.isSharedDevice();
+    } catch (error) {
+      console.log(error);
+    }
   }
 
   getAccount = async () => {
     try{
-      var {
-        authority,
-        id,
-        tenantId,
-        username,
-        idToken,
-      } = await MSAL.getAccount();
+      var result = await MSAL.getAccount();
       this.setState({
-        aUsername: username,
-        aTenantId: tenantId,
-        aIdToken: idToken,
-        aAuthority: authority,
-        aId: id,
-        isSignedIn: true
+        isSignedIn: true,
+        account: result,
       });
     } catch (message) {
       console.log(message);
     }
   }
+
+  isSharedDevice = async () => {
+    try {
+      var sharedDevice = MSAL.isSharedDevice();
+      this.setState({
+        isSharedDevice: sharedDevice
+      })
+    } catch (message) {
+      console.log(message);
+    }
+  }
+
+  startSignIn = async () => {
+    try{
+      var result = await MSAL.signIn(this.state.scopesValue);
+      this.setState({
+        isSignedIn: true,
+        account: result,
+      });
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  startSignOut = async () => {
+    try {
+      var result = await MSAL.signOut();
+      this.setState({
+        account: null,
+        graphResult: null,
+        isSignedIn: false
+      });
+    } catch (exception) {
+      console.log(exception)
+    }
+  }
+
+  startGetGraphDataSilently = async () => {
+    try {
+      var result = await MSAL.acquireTokenSilent(this.state.scopesValue);
+      //make api call
+      var response = await fetch (
+        this.state.URLValue, {
+          method: 'GET',
+          headers: {
+            'Authorization': result.accessToken
+          }
+        }
+      );
+      var json = await response.json();
+      this.setState({
+        graphResult: json
+      });
+    } catch (exception) {
+      console.log(exception);
+    }
+  }
+
+  showResult = () => {
+   return (this.state.graphResult != null ? JSON.stringify(this.state.graphResult, null, 4) : "");
```

Comment:
This can be `return this.state.graphResult && JSON.stringify(this.state.graphResult, null, 4)`

Reply:
Done. && in this case means return the second expression if the first is considered truthy, and if not return the first?